src/plugins/data_source_management/public/components/drop_down_header/drop_down_header.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './drop_down_header.scss';
+import { EuiTitle, EuiFlexGroup, EuiFlexItem, EuiLink } from '@elastic/eui';
+import React from 'react';
+import { ApplicationStart } from 'opensearch-dashboards/public';
+import { DSM_APP_ID } from '../../plugin';
+
+interface DataSourceOptionItemProps {
+  totalDataSourceCount: number;
+  activeDataSourceCount?: number;
+  application?: ApplicationStart;
+}
+
+export const DataSourceDropDownHeader: React.FC<DataSourceOptionItemProps> = ({
+  activeDataSourceCount,
+  totalDataSourceCount,
+  application,
+}) => {
+  const dataSourceCounterPrefix = totalDataSourceCount === 1 ? 'DATA SOURCE' : 'DATA SOURCES';
+  const dataSourceCounter =
+    activeDataSourceCount !== undefined
```

Comment:
Should we use `activeDataSourceCount != null` to check `undefined` and `null` both?

Reply:
since bother activeCount and totalCount should be coming from `somOptionsArray.length()`,  Since .length on an array in JavaScript always returns a non-negative integer, we should be good to not do null check here